src/strategy/periodic.rs

```diff
@@ -41,15 +58,57 @@ impl StrategyPeriodic {
             n => log::trace!("periodic updates, weekly calendar length: {} minutes", n),
         };
 
-        let strategy = Self { schedule: calendar };
+        let strategy = Self {
+            schedule: calendar,
+            time_zone: tz,
+        };
         Ok(strategy)
     }
 
+    /// Get the time zone from `periodic` strategy config.
+    fn get_time_zone(cfg: &inputs::PeriodicInput) -> Fallible<Tz> {
+        let tz;
+        if &cfg.time_zone == "localtime" {
+            // Use `read_link` instead of `exists()` because we only want to check for
+            // the existence of the `/etc/localtime` symlink, not whether it points to
+            // a valid location. (`read_link` returns an error if symlink doesn't exist)
+            if let Err(_) = read_link(Path::new("/etc/localtime")) {
+                tz = Tz::named("UTC");
+            } else {
+                tz = Tz::local();
```

Comment:
After switching to `anyhow`, we should probably add some context to the error that `Tz::local()` might return if `/etc/localtime` is a symlink to a bad/non-existent `tzfile`.
Currently, if `/etc/localtime` points to bogus, all we see is the following in the journal:
```
[ERROR] critical error: failed to assemble configuration settings
[ERROR]  -> failed to validate update-strategy configuration
[ERROR]  -> No such file or directory (os error 2)
```
It may be difficult to identify what went wrong in the update-strategy config.